maxim/test_runs/test_run_builder.py

```
@@ -230,47 +277,170 @@ def __process_entry(
-                    usage=YieldedOutputTokenUsage(
-                        latency=workflow_output.latency,
-                        completion_tokens=0,
-                        prompt_tokens=0,
-                        total_tokens=0,
+            # Check if we need to use simulation endpoints (simulationConfig + local evaluators)
+            has_local_evaluators = any(isinstance(e, BaseEvaluator) for e in self._config.evaluators)
+            use_simulation_endpoints = self._config.simulation_config is not None and has_local_evaluators
+            
+            if use_simulation_endpoints:
+                if not test_run_id:
+                    raise ValueError("test_run_id is required for simulation endpoints")
```

Ruff (0.14.14)

Avoid specifying long messages outside the exception class

(TRY003)